sqlcapture/tests/helpers.go

```diff
@@ -1,165 +1,72 @@
 	t.Helper()
-
-	var buf = new(CaptureOutputBuffer)
-	buf.MergeBase = CopyState(*state)
-	var initState = CopyState(*state)
-	if err := sqlcapture.RunCapture(ctx, tb.GetDatabase(), catalog, &initState, buf); err != nil {
-		fmt.Fprintf(&buf.Snapshot, "\n========\n\nCapture Terminated With Error:\n\n    %s\n", err.Error())
-	}
-
-	var result, states = buf.Output()
-	if len(states) > 0 {
-		*state = states[len(states)-1]
-	}
-	return result, states
-}
-
-// ConfiguredCatalog is a test helper for constructing a ConfiguredCatalog from stream names
-func ConfiguredCatalog(ctx context.Context, t testing.TB, tb TestBackend, streamNames ...string) airbyte.ConfiguredCatalog {
-	// Perform discovery and construct a map from names to discovered streams
-	var discoveredCatalog, err = sqlcapture.DiscoverCatalog(ctx, tb.GetDatabase())
-	if err != nil {
-		t.Fatal(err)
-	}
-	var streams = make(map[string]airbyte.Stream)
-	for _, stream := range discoveredCatalog.Streams {
-		streams[strings.ToLower(stream.Name)] = stream
-	}
-
-	var catalog = airbyte.ConfiguredCatalog{}
-	for _, name := range streamNames {
-		stream, ok := streams[strings.ToLower(name)]
-		if !ok {
-			t.Fatalf("no stream named %q was discovered", name)
-		}
-		catalog.Streams = append(catalog.Streams, airbyte.ConfiguredStream{
-			Stream: stream,
-		})
-	}
-	return catalog
-}
-
-// VerifySnapshot loads snapshot content from a file and compares with the
-// actual result of a test. The snapshot filename is derived automatically
-// from the current test name, with an optional suffix in case a single
-// test needs multiple snapshots. In the event of a mismatch, a ".new"
-// file is written for ease of comparison/updating.
-func VerifySnapshot(t *testing.T, suffix string, actual string) {
-	t.Helper()
-
-	var snapshotDir = "testdata"
-	var snapshotFile = snapshotDir + "/" + t.Name()
-	if suffix != "" {
-		snapshotFile += "_" + suffix
-	}
-	snapshotFile += ".snapshot"
-
-	var snapBytes, err = os.ReadFile(snapshotFile)
-	// Nonexistent snapshots aren't an error, because when adding a
-	// new test we'd like it to produce a "snapshot.new" file for us
-	// and the empty string won't match the expected result anyway.
-	if err != nil && !os.IsNotExist(err) {
-		t.Fatalf("error reading snapshot %q: %v", snapshotFile, err)
-	}
-
-	if actual == string(snapBytes) {
-		return
-	}
-
-	// Locate the first non-matching line and log it
-	var actualLines = strings.Split(actual, "\n")
-	var snapshotLines = strings.Split(string(snapBytes), "\n")
-	for idx := 0; idx < len(snapshotLines) || idx < len(actualLines); idx++ {
-		var x, y string
-		if idx < len(snapshotLines) {
-			x = snapshotLines[idx]
-		}
-		if idx < len(actualLines) {
-			y = actualLines[idx]
-		}
-		if x != y {
-			t.Errorf("snapshot %q mismatch at line %d", snapshotFile, idx)
-			t.Errorf("old: %s", x)
-			t.Errorf("new: %s", y)
-			break
+	var captureCtx, cancelCapture = context.WithCancel(ctx)
+	const shutdownDelay = 200 * time.Millisecond
```

Comment:
This seems like a pretty short delay - sounds like you've tested it and it works reliably though. Might be something we need to adjust if it proves flaky in the future.

Reply:
Short/long is all relative -- 200ms is enough time for a _fairly long_ network round trip, and at least for local tests (like in our CI builds) where the target database is running as a Docker container on the same machine this appears to be plenty generous.

So empirically this works, and we want it to be as short as possible because this is a constant overhead added to every capture of every test in the automated suite.

That said it might fail if I tried to do another round of the whole "throughput profiling under load" thing. But it shouldn't be too hard to increase the timeout if that becomes an issue.